### DISCUSSION TITLE
vitest and svelte component's onMount

### DISCUSSION THREAD

**Discussion by csandeep** [vitest-dev/vitest#3710] [Q&A] [answered]
I'm trying to use Svelte to create a simple reactive component. The component loads data from an api server `onMount` and updates a reactive value (which updates a html element).

I have a simple vitest that renders the component and verifies the value of the html element. However while running under vitest the `onMount` is never called and hence the api call is never made. What am I missing ?

**`Component.svelte`**:

```
<script>
	import { onMount } from 'svelte';

	export let name = 'world';

	onMount(async () => {
		console.log('chat onMount event!');
		const response = await fetch('http://localhost:8081/api');
        // for this example, assume name returned by api is FOO
		name = data.name;
	});

</script>

<div id="#element">
	<b> Hello {name}</b>
</div>
```

**`index.test.js`**:

```
import { expect, test } from 'vitest';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/svelte';
import Component from '../src/lib/Component.svelte';

test('should render', () => {
	render(Component);

	const heading = screen.getByText('Hello FOO');
	expect(heading).toBeInTheDocument();
});
```

**Comment by csandeep** [comment]
Seems to be related to https://github.com/testing-library/svelte-testing-library/issues/222 , still not sure if its a vitest issue or a issue with `testing-library/svelte-testing-library` -- happy to close this if anyone feels otherwise.

**Comment by AriPerkkio** [comment, marked as answer]
There's more discussion about this at https://github.com/vitest-dev/vitest/issues/2834. There are various work-arounds mentioned on that issue.
